src/frontend/src/lib/types/balance.ts

```diff
@@ -0,0 +1,7 @@
+import { BigNumber } from '@ethersproject/bignumber';
+
+export type Balance = BigNumber;
+
+export type NullableBalance = Balance | null;
+
+export type OptionalNullableBalance = NullableBalance | undefined;
```

Comment:
Have you try to have a single type `export type OptionBalance = Balance | null | undefined`? How does that look?

A bit an opiniated feedback but I needed few reads to process types like `Exclude<OptionalNullableBalance, null>` I have to say. It's early morning. 😆

Reply:
Ahahahahah have a look at last commit, I removed the `NullableBalance` type